src/main/java/com/thebois/models/beings/Colony.java

```diff
@@ -30,13 +28,12 @@
      * Creates a colony and fills it with pawns in the provided open positions.
      *
      * @param vacantPositions Positions in the world where pawns can be created.
-     * @param pathFinder      The pathfinder that the pawns will use.
      */
-    public Colony(final Iterable<Position> vacantPositions, final IPathFinder pathFinder) {
+    public Colony(final Iterable<Position> vacantPositions) {
         final Collection<IBeing> pawns = new ArrayList<>();
-        final Random random = new Random();
         for (final Position vacantPosition : vacantPositions) {
-            pawns.add(new Pawn(vacantPosition, vacantPosition, random, pathFinder));
+            final AbstractRole role = RoleFactory.idle();
+            pawns.add(new Pawn(vacantPosition, this, role));
```

Comment:
`this`?

Reply:
`Colony` implements `IInventory` and the `Pawn` needs access to the colony inventory.

Reply:
There is a `getInventory` in colony, couldn't Actions use that and only give it to the pawn when interactions are needed?

Reply:
The `HarvestAction` only has access to an `IActionPerformer`. In `IActionPerformer`, there is an `addItem` method. Since it is `AbstractBeing` which implements `IActionPerformer`, it has to implement `addItem`. Since `AbstractBeing` doesn't have an inventory of its own, it needs the inventory injected from `Colony`.

Reply:
But, once we've merged #83 into master, `AbstractBeing` will have its own inventory. So it'll no longer need to have it injected from `Colony`.